lib/change_stream.js

```diff
@@ -179,10 +179,27 @@ class ChangeStream extends EventEmitter {
     }
 
     // Tidy up the existing cursor
-    var cursor = this.cursor;
-    ['data', 'close', 'end', 'error'].forEach(event => this.cursor.removeAllListeners(event));
-    delete this.cursor;
-    return cursor.close(callback);
+    const cursor = this.cursor;
+
+    if (callback) {
```

Comment:
I don't think you need to add this code checking whether to return a `Promise` or not. The `Cursor` base class that `ChangeStreamCursor` extends already [does this for you](https://github.com/mongodb/node-mongodb-native/blob/master/lib/cursor.js#L882). The reason `return cursor.close(callback);` works is that if `callback` was not provided, it will automatically return a promise.

Reply:
That's true, but the problem is with [L189](https://github.com/mongodb/node-mongodb-native/pull/2084/files#diff-30861ceb96e5170099331504d187951eR189), which needs to behave differently for callbacks and promises.

Reply:
Oh I see one problem with my approach, you wouldn't return a rejected Promise if there was an error. Okay, let's leave it alone for now